cmd/security-agent/subcommands/start/command.go

```diff
@@ -289,12 +291,6 @@ func RunAgent(ctx context.Context, log log.Component, config config.Component, t
 		return errNoAPIKeyConfigured
 	}
 
-	err = manager.ConfigureAutoExit(ctx, config)
```

Comment:
The context (`ctx`) is not the same as the one in autoexit component which is `pkgcommon.GetMainCtxCancel()`. Is that correct?

Reply:
It seems that `pkgcommon.GetMainCtxCandel()` is returning `context.WithCancel(context.Background())`

https://github.com/DataDog/datadog-agent/blob/ea5bc8bfd62cfd9f11920ea3e4f6b42369e23fe6/pkg/util/common/context.go#L26-L32

Which (if I understand well) is the same as what is provided to `RunAgent()` function in the `start()` function:

https://github.com/DataDog/datadog-agent/blob/ecc60d202478065bde3789dfad60951464c6a4e1/cmd/security-agent/subcommands/start/command.go#L214-L218

Finally the `cancel()` function is called when the `start` agent end (with defer), in `StopAgent()` function:
https://github.com/DataDog/datadog-agent/blob/ecc60d202478065bde3789dfad60951464c6a4e1/cmd/security-agent/subcommands/start/command.go#L356-L358

Am I missing something, is it different ?

Reply:
Can you check with the security team that this change is correct as you have one instance instead of two?

Reply:
I asked @paulcacheux and everything looks good to him.